www/searchutil.php

```diff
@@ -568,6 +570,52 @@ function doSearch($db, $term, $searchType, $sortby, $limit, $browse)
                 }
                 break;
 
+            case 'rounded_median_time_in_minutes':
+                // we need to join the gametimes mv table for this query    
+                if (!isset($extraJoins[$col])) {
+                    $extraJoins[$col] = true;
+                    $tableList .= " inner join gametimes_mv "
+                                  . "on games.id = gametimes_mv.gameid";
+                }
+
+                // numeric range match
+                if ($txt == "")
+                    $expr = "$col is null";
+                else if (preg_match("/^([0-9.]+)-([0-9.]+)$/", $txt, $m))
+                    $expr = "$col >= '{$m[1]}' AND $col <= '{$m[2]}'";
+                else if (preg_match("/^([0-9.]+)[+-]$/", $txt, $m))
+                    $expr = "$col >= '{$m[1]}'";
+                else if (preg_match("/^-([0-9.]+)$/", $txt, $m))
+                    $expr = "$col <= '{$m[1]}'";
+                else if (preg_match("/^[0-9.]+$/", $txt))
+                    $expr = "$col = '$txt'";
+                else
+                    $expr = "$col = '" . mysql_real_escape_string($txt, $db) . "'";
+                break;  
+
+            case 'rounded_median_time_in_minutes/60':
+                // we need to join the gametimes mv table for this query    
+                if (!isset($extraJoins[$col])) {
+                    $extraJoins[$col] = true;
+                    $tableList .= " inner join gametimes_mv "
+                                  . "on games.id = gametimes_mv.gameid";
+                }
+
+                // numeric range match
+                if ($txt == "")
+                    $expr = "$col is null";
+                else if (preg_match("/^([0-9.]+)-([0-9.]+)$/", $txt, $m))
+                    $expr = "$col >= '{$m[1]}' AND $col <= '{$m[2]}'";
+                else if (preg_match("/^([0-9.]+)[+-]$/", $txt, $m))
+                    $expr = "$col >= '{$m[1]}'";
+                else if (preg_match("/^-([0-9.]+)$/", $txt, $m))
+                    $expr = "$col <= '{$m[1]}'";
+                else if (preg_match("/^[0-9.]+$/", $txt))
+                    $expr = "$col = '$txt'";
+                else
+                    $expr = "$col = '" . mysql_real_escape_string($txt, $db) . "'";
+                break;  
+
```

Comment:
It's a bummer to have so much code repetition here, but I don't really have a cleverer idea. Maybe @salty-horse would have a suggestion?

Reply:
I did a fall through to the next code block--you can see what you think.